aws/resource_aws_emr_cluster.go

```diff
@@ -661,13 +677,24 @@ func resourceAwsEMRClusterCreate(d *schema.ResourceData, meta interface{}) error
 			InstanceCount: aws.Int64(int64(m["instance_count"].(int))),
 			InstanceRole:  aws.String(emr.InstanceRoleTypeMaster),
 			InstanceType:  aws.String(m["instance_type"].(string)),
-			Market:        aws.String(emr.MarketTypeOnDemand),
```

Comment:
To maintain backwards compatibility, we should be able to leave this default then adjust the below logic leaving the original `bid_price` handling and adding `market` handling:

```go
if v, ok := m["bid_price"]; ok && v.(string) != "" {
	instanceGroup.BidPrice = aws.String(v.(string))
	instanceGroup.Market = aws.String(emr.MarketTypeSpot)
}

if v, ok := m["market"]; ok {
	instanceGroup.Market = aws.String(v.(string))
}
```

Same applies to `core_instance_group` changes below.